File: figures_suppl/Figure_S12_other_params_BGS/abc_bgs/scripts/subset_table_and_separate_params.R

library(reticulate); pd <- import("pandas")
#library(data.table)

source("scripts/rfunctions.R")

# read arguments from command line
args = commandArgs(trailingOnly=TRUE)

# explify args
transformer_file = args[1]
sumstat_file = args[2]
outfile_prefix = args[3]

# define output file
outfile_sumstat = paste0(outfile_prefix, ".sumstat", collapse="")

# read files 
transformer <- read.table(transformer_file, head=F, sep="\t", as.is=T)
sumstats <- pd$read_pickle(sumstat_file)

# save.image("/netscratch/dep_tsiantis/grp_laurent/struett/temp_tsabc/mchoice/tsabc3/TEMP.RData");
# stop("#######################################################################")
# load("TEMP.RData")

# explify names of columns to transform
colnames_to_be_transformed = transformer[,1]

# subset wanted sumstats
sumstats_of_interest <- dplyr::select(sumstats,
	all_of(colnames_to_be_transformed))

# extract params
params_all <- obtain_columns_with_wanted_sumstats(sumstats, "param")

# write subsetted data frames (params, sumstats) into separate files as txt
write.table(cbind(params_all, sumstats_of_interest),
	outfile_sumstat, quote = F, sep = "\t", row.names=F)

File: scripts/tsabc_with_masking/scripts/subset_table_and_separate_params.py
#!/netscratch/dep_tsiantis/grp_laurent/struett/conda/envs/tsabc3/bin/python

"""
reads transformer file, sumstat file and subsets the sumstats to the one chosen
for the transformer, add the params also to that file and write it down into a 
new unzipped .txt file
"""

import sys
import pandas as pd

transformer_file = sys.argv[1]
sumstat_file = sys.argv[2]
outfile_prefix = sys.argv[3]

# define outfile
outfile_sumstat = outfile_prefix + ".sumstat"

# read files
transformer = pd.read_csv(transformer_file, sep="\t", header=None)

if not sumstat_file.endswith(".txt"):
    sumstat = pd.read_pickle(sumstat_file)
else:
    sumstat = pd.read_csv(sumstat_file, sep="\t")

# subset columns to wanted stats
colnames_to_be_transformed = [*transformer[0]]
sumstats_of_interest = sumstat[colnames_to_be_transformed]

# extract params
params_all = sumstat.filter(regex="^param_")

# finalize data frame, params + sumstats
subsetted_table = pd.concat([params_all, sumstats_of_interest], axis=1)

# write to file
subsetted_table.to_csv(outfile_sumstat, sep="\t", index=False)
